### PR TITLE
move common code into coverlib

### PR DESCRIPTION
**Pull request by RaduBerinde**
#### separate line counts structure into coverlib

Separate the line counts structure into coverlib.

In gocover2json we now retain the maximum count if a line is part of
multiple code blocks (consistent with the bazel gocover to lcov code).

We also fix the license comments, they were always intended to be the
standard Apache.

#### move import/export code to coverlib

Add data structure for profiles and move logic to import and export
from various formats into coverlib.

This will make it very easy to introduce new tools (a gocover merger
and converter to LCOV will be needed for Pebble coverage).

- - -
This change is [Reviewable](https://reviewable.io/reviews/cockroachdb/code-cov-utils/2)

### CODE REVIEW

**Comment by RaduBerinde** [comment]
Hi @srosenberg let me know if you prefer I just push changes without review (I have more coming). Perhaps it's easier to just take a look at the final state of things.

File: coverlib/export_codecov_json.go [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+//	      "1": 0,      # line 1 missed
+//	      "2": 1,      # line 2 hit once
+//	      "7": 5       # line 7 hit 5 times
+//	    }
+//	  }
+//	}
+func ExportCodecovJson(p *Profiles, writer io.Writer) error {
+	// The output schema is odd, in that each line is a separate attribute
+	// (instead of being part of an array). This makes it hard to use Go's json
+	// machinery; we just produce the output directly.
```
Comment (renatolabs) [inline coverlib/export_codecov_json.go]: IMO, we should still use the json stdlib package. The output can be the JSON encoding of the following type (untested, but you get the point):

```go
type FileCoverage map[string]int // line number to hit count

type Data struct {
    Coverage map[string]FileCoverage `json:"coverage"` // filename to coverage
}
```

Comment (RaduBerinde) [reply coverlib/export_codecov_json.go]: Oh, awesome, I'll give that a shot!

Comment (RaduBerinde) [reply coverlib/export_codecov_json.go]: Done. I found in the docs that it even converts integer keys to strings automatically (unfortunately the resulting fields are still sorted as strings not integers).

**Review by renatolabs — APPROVED** [review]

**Comment by RaduBerinde** [comment]
TFTR!

**Pull request merged**